src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/UserChannel.cs

```diff
@@ -0,0 +1,149 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using System.Collections.Concurrent;
+using System.Reactive.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.Extensions.Logging;
+using MorganStanley.ComposeUI.Messaging;
+
+namespace DesktopAgent
+{
+    public class UserChannel
```

Comment:
No, but I really should name it better exactly because of this confusion :) When I originally created the class, I was also considering implementing the interface, but that interface is for calling the API not for handling the incoming messages.

Reply:
Because this is the backend implementation, not a client API, OK got it :)